### PR TITLE
Decode a class

### PR DESCRIPTION
**Pull request by Owusu-Desmond**
in this pull request, I did the following:

- [ ] I created a new file called `nameable.rb` that creates a nameable class

- [ ] I added a new files called `capitalize_decorator.rb` and `trimmer_decorator.rb`

- [ ] Turn your Person class to Nameable

- [ ] Create new file called `main.rb` to test the code

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+class CapitalizeDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
```
Comment (codepantha) [inline capitalize_decorator.rb]: - [ ] You have created CapitalizeDecorator 👏 Well done. However, Capitalize Decorator should inherit from a Base Decorator that you have not created yet. Please create a base decorator which inherits from Nameable according to project specifications 👍

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class TrimmerDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```
Comment (codepantha) [inline trimmer_decorator.rb]: - [ ] According to the project specifications, your `TrimmerDecorator `class should inherit from a base decorator that you have not created yet. Please create the base decorator 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @Owusu-Desmond 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @Owusu-Desmond , 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive PR.
:heavy_check_mark: Well implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "fix bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**